test/e2e/tests/imagelist_include_nodes/eraser_test.go

```diff
@@ -0,0 +1,195 @@
+				if err != nil {
+					return false, err
+				}
+
+				return len(nodeList.Items) == 1, nil
+			}, wait.WithTimeout(time.Minute))
+			if err != nil {
+				t.Errorf("error while waiting for selector%s to be added to node\n%#v", util.FilterNodeSelector, err)
+			}
+
+			resultDeployment := appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{Name: util.Nginx, Namespace: cfg.Namespace()},
+			}
+
+			if err = wait.For(
+				conditions.New(cfg.Client().Resources()).DeploymentConditionMatch(&resultDeployment, appsv1.DeploymentAvailable, corev1.ConditionTrue),
+				wait.WithTimeout(time.Minute*3),
+			); err != nil {
+				t.Error("deployment not found", err)
+			}
+
+			return context.WithValue(ctx, util.Nginx, &resultDeployment)
+		}).
+		Assess("Node(s) successfully included", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			// delete deployment
+			client, err := cfg.NewClient()
+			if err != nil {
+				t.Error("Failed to create new client", err)
+			}
+
+			var pods corev1.PodList
+			err = client.Resources().List(ctx, &pods, func(o *metav1.ListOptions) {
+				o.LabelSelector = labels.SelectorFromSet(labels.Set{"app": util.Nginx}).String()
+			})
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			dep := ctx.Value(util.Nginx).(*appsv1.Deployment)
+			if err := client.Resources().Delete(ctx, dep); err != nil {
+				t.Error("Failed to delete the dep", err)
+			}
+
+			err = wait.For(util.ContainerNotPresentOnNode(util.FilterNodeName, util.Nginx), wait.WithTimeout(time.Minute*2))
+			if err != nil {
+				// Let's not mark this as an error
+				// We only have this to prevent race conditions with the eraser spinning up
+				t.Logf("error while waiting for deployment deletion: %v", err)
+			}
+
+			// deploy imageJob config
+			if err = util.DeployEraserConfig(cfg.KubeconfigFile(), "eraser-system", "../../test-data", "eraser_v1alpha1_imagelist.yaml"); err != nil {
+				t.Error("Failed to deploy image list config", err)
+			}
+
+			ctxT, cancel := context.WithTimeout(ctx, time.Minute)
+			defer cancel()
+
+			// ensure image is removed from filtered node.
+			util.CheckImageRemoved(ctxT, t, []string{util.FilterNodeName}, util.Nginx)
+
+			// Wait for the imagejob to be completed by checking for its nonexistence in the cluster
+			err = wait.For(util.ImagejobNotInCluster(cfg.KubeconfigFile()), wait.WithTimeout(time.Minute*2))
+			if err != nil {
+				t.Logf("error while waiting for imagejob cleanup: %v", err)
+			}
+
+			clusterNodes := util.GetClusterNodes(t)
+			clusterNodes = util.DeleteStringFromSlice(clusterNodes, util.FilterNodeName)
+
+			// the imagejob has done its work, so now we can check the node to make sure it didn't remove the images from the remaining nodes
+			util.CheckImagesExist(ctx, t, clusterNodes, util.Nginx)
+
+			return ctx
+		}).
+		Teardown(func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
```

Comment:
We don't need the `Teardown` anymore, because the whole kind cluster gets destroyed after the test is run.